tests/Integration/Presentation/Presenters/DonationConfirmationHtmlPresenterTest.php

```diff
@@ -21,70 +22,93 @@ class DonationConfirmationHtmlPresenterTest extends \PHPUnit\Framework\TestCase
 	private const STATUS_BOOKED = 'status-booked';
 	private const STATUS_UNCONFIRMED = 'status-unconfirmed';
 
+	private const UPDATE_TOKEN = 'update_token';
+	private const ACCESS_TOKEN = 'access_token';
+	private const DONATION_ID = 42;
+
 	public function testWhenPresenterRenders_itPassedParamsToTemplate(): void {
-		$twig = $this->getMockBuilder( TwigTemplate::class )->disableOriginalConstructor()->getMock();
-		$pageSelector = $this->getMockBuilder( SelectedConfirmationPage::class )->disableOriginalConstructor()->getMock();
+		$expectedParameters = $this->getExpectedRenderParams();
+		$expectedParameters['donation']['status'] = self::STATUS_BOOKED;
 
-		$twig->expects( $this->once() )
-			->method( 'render' )
-			->with( $this->getExpectedRenderParams( self::STATUS_BOOKED ) );
+		$presenter = new DonationConfirmationHtmlPresenter(
+			$this->newTwigTemplateMock( $expectedParameters ),
+			new FakeUrlGenerator()
+		);
+
+		$donation = ValidDonation::newBookedAnonymousPayPalDonation();
+		$donation->assignId( self::DONATION_ID );
 
-		$presenter = new DonationConfirmationHtmlPresenter( $twig );
 		$presenter->present(
-			ValidDonation::newBookedAnonymousPayPalDonation(),
-			'update_token',
-			$pageSelector,
+			$donation,
+			self::UPDATE_TOKEN,
+			self::ACCESS_TOKEN,
+			$this->newSelectedConfirmationPage(),
 			$this->newPiwikEvents()
 		);
 	}
 
-	private function getExpectedRenderParams( string $mappedStatus ): array {
+	private function getExpectedRenderParams(): array {
 		return [
 			'template_name' => '',
 			'templateCampaign' => '',
 			'donation' => [
-				'id' => null,
-				'status' => $mappedStatus,
+				'id' => self::DONATION_ID,
 				'amount' => 13.37,
 				'interval' => 3,
 				'paymentType' => 'PPL',
 				'optsIntoNewsletter' => false,
 				'bankTransferCode' => '',
 				'creationDate' => ( new \DateTime() )->format( 'd.m.Y' ),
 				'cookieDuration' => '15552000',
-				'updateToken' => 'update_token'
+				'updateToken' => self::UPDATE_TOKEN
 			],
-			'person' => [ ],
-			'bankData' => [ ],
-			'initialFormValues' => [ ],
+			'person' => [],
+			'bankData' => [],
+			'initialFormValues' => [],
 			'piwikEvents' => [
 				[ 'setCustomVariable', 1, 'Payment', 'some value', PiwikEvents::SCOPE_VISIT ],
 				[ 'trackGoal', 4095 ]
-			]
+			],
+			'commentUrl' => 'https://such.a.url/AddCommentPage?donationId=42&updateToken=update_token&accessToken=access_token'
```

Comment:
Is routing not set up in the test environment? FWIW this shows the route name, not the path - not necessarily bad but surprising.

Reply:
This test does not use an "environment". See line 35 for the fake service used. All we care for in the test here is that the parameters are passed to this service, and this is done.

Reply:
Soo much scrolling.